.github/workflows/create-release-tag.yml

```diff
@@ -0,0 +1,198 @@
+            PKG_RC_NUMBER="${BASH_REMATCH[1]}"
+          else
+            PKG_RC_NUMBER="0"
+          fi
+          echo "package_version=$PACKAGE_VERSION" | tee -a $GITHUB_OUTPUT
+          echo "base_version=$BASE_VERSION" | tee -a $GITHUB_OUTPUT
+          echo "rc_number=$PKG_RC_NUMBER" | tee -a $GITHUB_OUTPUT
+
+      - name: Get latest RC tag
+        id: get-latest-rc
+        run: |
+          # Fetch all tags
+          git fetch --tags || true
+          
+          BASE_VERSION="${{ steps.pkg-version.outputs.base_version }}"
+          
+          # Get the latest RC tag for this base version (format: vX.Y.Z-rcN)
+          LATEST_RC_TAG=$(git tag -l "v${BASE_VERSION}-rc*" | sort -V | tail -1)
+          
+          # Extract RC number if a matching tag exists; accept optional suffix like -signed
+          if [[ "$LATEST_RC_TAG" =~ ^v${BASE_VERSION}-rc([0-9]+)(-.+)?$ ]]; then
+            LATEST_RC_NUMBER="${BASH_REMATCH[1]}"
+          else
+            LATEST_RC_NUMBER="0"
+          fi
+          
+          echo "latest_rc_tag=$LATEST_RC_TAG" | tee -a $GITHUB_OUTPUT
+          echo "latest_rc_number=$LATEST_RC_NUMBER" | tee -a $GITHUB_OUTPUT
+          echo "base_version=$BASE_VERSION" | tee -a $GITHUB_OUTPUT
+
+      - name: Calculate new version
+        id: calculate-version
+        run: |
+          STRATEGY="${{ inputs.version_strategy }}"
+          BASE_VERSION="${{ steps.pkg-version.outputs.base_version }}"
+          
+          # Set tag suffix, commit message suffix, delete tag flag, and message prefix based on strategy
+          if [ "$STRATEGY" = "production" ]; then
+            TAG_SUFFIX="-signed"
+            COMMIT_SUFFIX="[skip release]"
+            DELETE_EXISTING_TAG="false"
+            TAG_MESSAGE_PREFIX="Production Release"
+          else
+            TAG_SUFFIX=""
+            COMMIT_SUFFIX=""
+            DELETE_EXISTING_TAG="true"
+            TAG_MESSAGE_PREFIX="Feature Release"
+          fi
+          
+          if [ "$STRATEGY" = "production" ]; then
+            # Production: always increment from latest RC
+            LATEST_RC_NUMBER=${{ steps.get-latest-rc.outputs.latest_rc_number }}
+            NEW_RC_NUMBER=$((LATEST_RC_NUMBER + 1))
+            NEW_VERSION="$BASE_VERSION-rc$NEW_RC_NUMBER"
+            NEW_TAG="v$NEW_VERSION$TAG_SUFFIX"
+          elif [ "$STRATEGY" = "increment" ]; then
+            # Increment: increment from latest RC
+            LATEST_RC_NUMBER="${{ steps.get-latest-rc.outputs.latest_rc_number }}"
+            NEW_RC_NUMBER=$(( LATEST_RC_NUMBER + 1 ))
+            NEW_VERSION="$BASE_VERSION-rc$NEW_RC_NUMBER"
+            NEW_TAG="v$NEW_VERSION$TAG_SUFFIX"
+          else
+            # Override: use the same RC as package.json
+            NEW_RC_NUMBER="${{ steps.pkg-version.outputs.rc_number }}"
+            NEW_VERSION="${{ steps.pkg-version.outputs.package_version }}"
+            NEW_TAG="v$NEW_VERSION$TAG_SUFFIX"
+          fi
```

Comment:
I'd put this in the "elif" instead, it's safer to have "increment" behaviour as default for "else"